### ISSUE TITLE
Move the cube generation code to a thread

### ISSUE THREAD

**Issue by saifulkhan** [ScottishCovidResponse/rampvis-api#21]
It blocks the server statup 

```
2021-11-19 16:58:09.523 | INFO     | apscheduler.schedulers.background:start:35 - Scheduler started
2021-11-19 16:58:09.523 | DEBUG    | apscheduler.schedulers.blocking:_main_loop:32 - Looking for jobs to run
2021-11-19 16:58:09.524 | INFO     | app.controllers.timeseries_sim_search_controller:<module>:73 - Timeseries similarity agent starts. Will run immediately now and every 1am.
2021-11-19 16:58:09.524 | DEBUG    | apscheduler.schedulers.blocking:_main_loop:32 - Next wakeup is due at 2021-11-20 01:00:00+00:00 (in 28910.475516 seconds)
2021-11-19 16:58:13.251 | INFO     | app.algorithms.sim_search.cube_functions:cube_master:91 - cube_master will take a few minutes...
2021-11-19 16:58:13.256 | INFO     | app.algorithms.sim_search.cube_functions:cube_master:95 - Afghanistan
2021-11-19 16:58:15.099 | INFO     | app.algorithms.sim_search.cube_functions:cube_master:95 - Africa
2021-11-19 16:58:16.766 | INFO     | app.algorithms.sim_search.cube_functions:cube_master:95 - Albania
```

**Comment by phongvis** [comment]
I just moved all agents (data download, sensitivity, time series search) to threads so that they don't block the server. I also deleted data files that can be downloaded and git-ignored them. Please check if the changes cause any issues. @tunagonen @ErikRZH

**Comment by ErikRZH** [comment]
A concern regarding this change. It results in the following behaviour
On startup: `sensitivity_anlysis_agent.py` agent runs before `raw.json` is downloaded (as they are on separate threads), thus `processed.json` is not generated on startup anymore. So if `processed.json` is not already present on the system the plot needing based on this file will be broken until 1 am when the server restarts.

**Comment by phongvis** [comment]
@saifulkhan when you deploy an update, do you checkout the entire repo and all existing data gone?

**Comment by saifulkhan** [comment]
In an "ideal" deployment scenario, yes everything should be fresh. I have not deployed the recent code yet. I am waiting for the recent commits to be tested.

**Comment by phongvis** [comment]
ok, so data download should be the first thing to do to get data. I will make a change to move the data download agent to run on the same thread and block other actions. Other agents are still in separate threads and run after data is downloaded.

btw, I fixed #24 , please deploy. Also, I post an error I got with latest UI code there.
